### ISSUE TITLE
SD card is not properly recognized under Linux (possible fix attached)

### ISSUE THREAD

**Issue by fmauch** [winzkigermany/hoerbert-software#12]
Hi,

On my Linux system the SD card is not properly recognized inside the software (tested version 2.1.5 built from source and version 3.1.3 from AppImage on Archlinux with systemd=253). As the [Software FAQ](https://www.hoerbert.com/service/downloads-und-bespielen/) already notes the workaround of specifying the mountpoint manually, this seems to be a known and common problem.

When skimming through the source code, it seems to fail in the [`isRemovable`](https://github.com/winzkigermany/hoerbert-software/blob/ba74b1cb5abb612b062c70e231ace7683961cb6c/hoerbert/devicemanager.cpp#L73) method of the `DeviceManager`. I noticed, that the check for `ID_USB_DRIVER='usb-storage'` indeed is not in the output of the `udevadm info` call regarding my mounted drive. My output of `udevadm info --query=property --export --name=/dev/mmcblk0` is

```
DEVPATH='/devices/pci0000:00/0000:00:02.1/0000:02:00.0/rtsx_pci_sdmmc.0/mmc_host/mmc0/mmc0:aaaa/block/mmcblk0'
DEVNAME='/dev/mmcblk0'
DEVTYPE='disk'
DISKSEQ='6'
MAJOR='179'
MINOR='0'
SUBSYSTEM='block'
USEC_INITIALIZED='1384769046'
ID_SERIAL='0x535809eb'
ID_NAME='SN64G'
ID_PATH='pci-0000:02:00.0-platform-rtsx_pci_sdmmc.0'
ID_PATH_TAG='pci-0000_02_00_0-platform-rtsx_pci_sdmmc_0'
ID_PART_TABLE_TYPE='dos'
MMC_TYPE='SD'
ID_DRIVE_FLASH_SD='1'
ID_DRIVE_MEDIA_FLASH_SD='1'
DEVLINKS='/dev/disk/by-path/pci-0000:02:00.0-platform-rtsx_pci_sdmmc.0 /dev/disk/by-id/mmc-SN64G_0x535809eb /dev/disk/by-diskseq/6'
TAGS=':systemd:'
CURRENT_TAGS=':systemd:'
```

When replacing the regex in the code with

```c++
    return output.contains(QRegExp("ID_DRIVE_FLASH_SD='1'", Qt::CaseSensitive));
```

the drive is properly recognized. I'm not sure whether that change would fix things on other systems, as well.

It would be nice to have the latest version here on Github, btw. It does not feel right to open a PR on a 3 year old version, which is why I created this issue instead.

**Comment by jmr** [comment]
It looks like checking `/sys/block/mmcblk0/removable` might be the preferred way. Does a path like that exist?

https://stackoverflow.com/a/73302717

> It would be nice to have the latest version here on Github, btw.

Someone figured out that the `feedback`  branch is the current one: https://github.com/winzkigermany/hoerbert-software/tree/feedback

**Comment by winzkigermany** [comment]
Yes, for historical reasons, the feedback branch is the most recent.
I've never changed that so far, "works for me" (TM). ;-)

Thank you for your feedback. Still I'm "slow to respond" and it will take me a while to incorporate your research.

**Comment by fmauch** [comment]
Glad to hear that the `feedback` branch is actually the recent version :-)

@jmr `cat /sys/block/mmcblk0/removable` contains a `0` for me, so this seems not to be working that reliably?

**Comment by jmr** [comment]
> `cat /sys/block/mmcblk0/removable` contains a `0` for me

Maybe report this as a bug to your distribution. Do `fixed` and `unknown` files exist, and what do they contain?

**Comment by fmauch** [comment]
> Maybe report this as a bug to your distribution. Do `fixed` and `unknown` files exist, and what do they contain?

`fixed` and `unknown` do not exist.

Who is responsible for filling this? Is this also coming directly from systemd?

**Comment by jmr** [comment]
> fixed and unknown do not exist.

I just saw those strings here, so I thought maybe there were files for them, too:

https://mjmwired.net/kernel/Documentation/ABI/testing/sysfs-devices-removable

> Who is responsible for filling this? 

I believe `/sys` comes from the kernel.

**Comment by rogerkk** [comment]
FWIW, I'm experiencing the same bug on Ubuntu 22.04.5 LTS.

Here's my output from `udevadm info --query=property --export --name=/dev/mmcblk0` in case it's of any use:

```
DEVPATH='/devices/pci0000:00/0000:00:1c.5/0000:3b:00.0/rtsx_pci_sdmmc.0/mmc_host/mmc0/mmc0:5048/block/mmcblk0'
DEVNAME='/dev/mmcblk0'
DEVTYPE='disk'
DISKSEQ='85'
MAJOR='179'
MINOR='0'
SUBSYSTEM='block'
USEC_INITIALIZED='73861169255'
ID_SERIAL='0xda23bff6'
ID_NAME='SD04G'
ID_PATH='pci-0000:3b:00.0-platform-rtsx_pci_sdmmc.0'
ID_PATH_TAG='pci-0000_3b_00_0-platform-rtsx_pci_sdmmc_0'
ID_PART_TABLE_UUID='75563031'
ID_PART_TABLE_TYPE='dos'
MMC_TYPE='SD'
ID_DRIVE_FLASH_SD='1'
ID_DRIVE_MEDIA_FLASH_SD='1'
DEVLINKS='/dev/disk/by-id/mmc-SD04G_0xda23bff6 /dev/disk/by-path/pci-0000:3b:00.0-platform-rtsx_pci_sdmmc.0'
TAGS=':systemd:'
CURRENT_TAGS=':systemd:'
```

I'm also getting `0` from `cat /sys/block/mmcblk0/removable`. No `fixed` or `unknown` here either.
